test/e2e/pivoting_test.go

```diff
@@ -0,0 +1,263 @@
+package e2e
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	bmo "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+	capm3 "github.com/metal3-io/cluster-api-provider-metal3/api/v1alpha4"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+
+	//"sigs.k8s.io/cluster-api/test/e2e/internal/log"
+	dockerTypes "github.com/docker/docker/api/types"
+	docker "github.com/docker/docker/client"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/config"
+	framework "sigs.k8s.io/cluster-api/test/framework"
+	"sigs.k8s.io/cluster-api/test/framework/clusterctl"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func pivoting() {
+	By("Remove Ironic containers in the source cluster")
+	ephemeralCluster := os.Getenv("EPHEMERAL_CLUSTER")
+	if ephemeralCluster == "kind" {
+		removeIronicContainers()
+	} else {
+		removeIronicDeployment()
+	}
+
+	By("Create Ironic namespace")
+	targetCluster := bootstrapClusterProxy.GetWorkloadCluster(ctx, namespace, clusterName)
+	targetClusterClientSet := targetCluster.GetClientSet()
+	ironicNamespaceEnv := "IRONIC_NAMESPACE"
+	ironicNamespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: os.Getenv(ironicNamespaceEnv),
+		},
+	}
+	_, err := targetClusterClientSet.CoreV1().Namespaces().Create(ironicNamespace)
+	Expect(err).To(BeNil(), "Unable to create the Ironic namespace")
+
+	By("Configure Ironic Configmap")
+	configureIronicConfigmap(true)
+
+	capm3Version := os.Getenv("CAPM3_VERSION")
+	if capm3Version == "v1alpha5" {
+		By("Add labels to BMO CRDs")
+		labelBMOCRDs(nil)
+		By("Install BMO")
+		installIronicBMO(targetCluster, "false", "true")
+	}
+
+	By("Initialize Provider component in target cluster")
+	clusterctl.Init(context.TODO(), clusterctl.InitInput{
```

Comment:
Since we have the global `ctx` variable, maybe it should be used here?
Although I must admit I don't see any reason or benefit for passing all this ctx around...

Reply:
Fixed! Thanks